frontend/src/views/student/tournament/ParticipantsTournamentView.vue

```diff
@@ -0,0 +1,187 @@
+                  text-color="white"
+                >
+                  {{ item.ranking }}
+                </v-chip>
+                <span v-else-if="item.ranking != 0">
+                  {{ item.ranking }}
+                </span>
+              </template>
+              <template v-slot:item.score="{ item }">
+                <v-chip v-if="isClosed" :color="getPercentageColor(item.score)">
+                  {{ item.score === '' ? 'NOT SOLVED' : item.score }}
+                </v-chip>
+                <v-chip v-else>
+                  {{ 'NOT AVAILABLE YET' }}
+                </v-chip>
+              </template>
+            </v-data-table>
+          </v-card>
+        </v-col>
+      </v-row>
+    </v-container>
+  </v-container>
+</template>
+
+<script lang="ts">
+import { Component, Prop, Vue } from 'vue-property-decorator';
+import Tournament from '@/models/user/Tournament';
+import SolvedQuiz from '@/models/statement/SolvedQuiz';
+import TournamentScore from '@/models/user/TournamentScore';
+import User from '@/models/user/User';
+import RemoteServices from '@/services/RemoteServices';
+
+@Component({})
+export default class ParticipantsTournament extends Vue {
+  @Prop({ type: String, required: true }) id!: number;
+
+  tournaments: Tournament[] = [];
+  closedTournaments: Tournament[] = [];
+  selectedTournament: Tournament | null = null;
+  isClosed: Boolean = false;
+  quizzes: SolvedQuiz[] = [];
+  tournamentScores: TournamentScore[] = [];
+  correctAnswers: number = 0;
+
+  headers: object = [
+    { text: 'Ranking', value: 'ranking', align: 'center' },
+    { text: 'Name', value: 'name', align: 'center' },
+    { text: 'Username', value: 'username', align: 'center' },
+    { text: 'Score', value: 'score', align: 'center' }
+  ];
+
+  async created() {
+    await this.$store.dispatch('loading');
+    try {
+      this.tournaments = await RemoteServices.getAllTournamentsForCourseExecution();
+      this.closedTournaments = await RemoteServices.getClosedTournamentsForCourseExecution();
+      this.tournaments.map(tournament => {
+        if (tournament.id == this.id) this.selectedTournament = tournament;
+      });
+      this.closedTournaments.map(tournament => {
+        if (tournament.id == this.id) this.isClosed = true;
+      });
+      if (this.selectedTournament)
+        for (const participant of this.selectedTournament.participants) {
+          let score;
+          if (this.isClosed) score = await this.getScore(participant);
+          else score = '';
+          this.tournamentScores.push(
+            new TournamentScore(participant, score, this.correctAnswers)
+          );
+        }
+      if (this.isClosed)
+        this.tournamentScores.sort((a, b) => this.sortByScore(a, b));
+      this.setRankings();
+    } catch (error) {
+      await this.$store.dispatch('error', error);
+    }
+    await this.$store.dispatch('clearLoading');
+  }
+
+  sortByScore(a: TournamentScore, b: TournamentScore) {
```

Comment:
this should be a method of TournamentScore

Reply:
Done